astropy/coordinates/funcs.py

```diff
@@ -176,8 +176,87 @@ def concatenate(coords):
         A single sky coordinate with its data set to the concatenation of all
         the elements in ``coords``
     """
-    from .sky_coordinate import SkyCoord
-
     if getattr(coords, 'isscalar', False) or not isiterable(coords):
         raise TypeError('The argument to concatenate must be iterable')
     return SkyCoord(coords)
+
+
+# global dictionary that caches repeatedly-needed info for get_constellation
+_constellation_data = {}
+def get_constellation(coord, short=False):
+    """
+    Determines the constellation(s) of the coordinates this `SkyCoord`
```

Comment:
Oops, yeah.  The need for this change (and a similar change in "Returns") also means the bit I suggested on astropy-dev probably isn't worth it for this... Will fix